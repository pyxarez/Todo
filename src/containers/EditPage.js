import React, { Component, PropTypes } from 'react';
import { connect } from 'react-redux';
import { toNumberKeyValues } from '../utils/helpers';

import Header from '../components/Header';
import NotFoundPage from '../components/NotFoundPage';

import EditTaskContainer from './EditTask';
import EditPageCategoryContainer from './EditPageCategory';
import Sidebar from './Sidebar';

import './EditPage.res/style.css';

export class EditPageContainer extends Component {
  static propTypes = {
    tasks: PropTypes.arrayOf(PropTypes.object.isRequired).isRequired,
    params: PropTypes.shape({
      categoryId: PropTypes.string.isRequired,
      taskId: PropTypes.string.isRequired,
      taskTitle: PropTypes.string.isRequired,
    }),
  }

  render() {
    const {
      tasks,
      params
    } = this.props;
    const {
      categoryId,
      taskId
    } = toNumberKeyValues(params);
    
    const task = tasks
      .find((task) => task.id === taskId);

    if (!task) return <NotFoundPage />;

    return (
      <div className="my-edit-page-component">
        <Header title={task.title} />
        <div className="main">
          <Sidebar
            categoryContainer={EditPageCategoryContainer}
            currentCategoryId={categoryId}
            currentTaskId={taskId}/>
          <EditTaskContainer
            task={task}
            categoryId={categoryId}/>
        </div>
      </div>
    );
  }
}

const mapStateToProps = ({ tasks }) => {
  return {
    tasks
  }
}

export default connect(mapStateToProps)(EditPageContainer);
