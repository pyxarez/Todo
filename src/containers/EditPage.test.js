import React from 'react';
import { shallow } from 'enzyme';

import { EditPageContainer } from './EditPage';

const setup = () => {
  const props = {
    tasks: [
      {
        id: 0,
        title: 'lol',
        isDone: false,
        description: 'troll'
      },
      {
        id: 1,
        title: 'Mishanya',
        isDone: true,
        description: 'woke up'
      }
    ],
    params: {
      categoryId: '0',
      taskId: '1',
      taskTitle: 'Task title'
    },
  };

  const wrapper = shallow(<EditPageContainer {...props}/>);

  return {
    wrapper,
    props
  }
};

describe('Containers :: EditPageContainer', () => {
  it('should render self', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should render NotFoundPage for lack of requested task', () => {
    const { props } = setup();
    const newProps = {
      ...props,
      params: {
        ...props.params,
        taskId: '3'
      }
    }
    const wrapper = shallow(<EditPageContainer {...newProps}/>);
    
    expect(wrapper).toMatchSnapshot();
  });
});
