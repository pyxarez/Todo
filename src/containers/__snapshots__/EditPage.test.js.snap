exports[`Containers :: EditPageContainer should render NotFoundPage for lack of requested task 1`] = `<NotFound />`;

exports[`Containers :: EditPageContainer should render self 1`] = `
<div
  className="my-edit-page-component">
  <Header
    title="Mishanya" />
  <div
    className="main">
    <Connect(Sidebar)
      categoryContainer={[Function]}
      currentCategoryId={0}
      currentTaskId={1} />
    <Connect(EditTaskContainer)
      categoryId={0}
      task={
        Object {
          "description": "woke up",
          "id": 1,
          "isDone": true,
          "title": "Mishanya",
        }
      } />
  </div>
</div>
`;
